### PR TITLE
EQL: Test infrastructure improvements

### PR DESCRIPTION
**Pull request by aleksmaus**
Update CommonEqlRestTestCase code to simplify making changes as requested.
Update EqlActionIT to simplify the test code as requested.
Replace Jackson parser with XContent in EqlActionIT.
Whitelist more EQL tests specs that are now supported.

Related to https://github.com/elastic/elasticsearch/issues/52942

### CODE REVIEW

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlActionIT.java [hunk 1]
```diff
@@ -7,60 +7,63 @@
...
+                // Adjust the structure of the document with additional event.category and @timestamp fields
+                // Add event.category field
+                HashMap<String, Object> objEvent = new HashMap<>();
+                objEvent.put("category", entry.get("event_type"));
+                entry.put("event", objEvent);
+
+                // Add @timestamp field
+                entry.put("@timestamp", entry.get("timestamp"));
+
+                //BytesReference bytes = XContentHelper.toXContent(entry, XContentType.JSON, false);
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlActionIT.java]: Leftover commented line?

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlActionIT.java [hunk 2]
```diff
@@ -108,19 +111,24 @@ public EqlActionIT(int num, EqlSpec spec) {
...
-        EqlSearchResponse response = new EqlSearchRequestBuilder(client(), EqlSearchAction.INSTANCE)
-            .indices(testIndexName).query(spec.query()).get();
-
-        List<SearchHit> events = response.hits().events();
-        assertNotNull(events);
-
+    private static long[] extractIds(List<SearchHit> events) {
         final int len = events.size();
         final long ids[] = new long[len];
         for (int i = 0; i < events.size(); i++) {
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlActionIT.java]: `len` defined above, but not actually used here.

**Review by astefan — APPROVED** [review]
Left two comments, but otherwise LGTM.

**Review by imotov — APPROVED** [review]
LGTM2

**Pull request merged**